js/customize-snapshots.js

```diff
@@ -104,15 +107,22 @@
 		 * @return {{}} Query vars for scroll, device, url, and autofocus.
 		 */
 		getStateQueryVars: function() {
-			var queryVars = {
+			var snapshot = this, currentTheme, queryVars;
+
+			queryVars = {
 				'autofocus[control]': null,
 				'autofocus[section]': null,
 				'autofocus[panel]': null
 			};
+			currentTheme = api.settings.theme.stylesheet;
 			queryVars.scroll = parseInt( api.previewer.scroll, 10 ) || 0;
 			queryVars.device = api.previewedDevice.get();
 			queryVars.url = api.previewer.previewUrl.get();
 
+			if ( ! api.state( 'activated' ).get() || snapshot.data.theme !== currentTheme ) {
+				queryVars.previewingTheme = true;
```

Comment:
Since this is going to be sent with the POST request, I think it should be `previewing_theme` instead. Then below in PHP, you'd read ` $query_vars['previewing_theme']`.